src/components/TextInput.jsx

```diff
@@ -180,31 +202,20 @@ export default class TextInput extends Component {
 
     if (!this.props.allowNewLine && value.match(newLineRegex) !== null) {
       value = value.replace(newLineRegex, ' ');
-
-      // controlled textarea must have value
-      if (this.state.textareaProps.value) {
-        this.setState({ textareaProps: { value: value } });
-        this.forceUpdate(this._triggerResize);
-      // uncontrolled textarea must be updated with value, but then released again
-      } else {
-        this.setState({ textareaProps: { value: value } });
-        this.forceUpdate(() => {
-          this._triggerResize();
-          this.setState({ textareaProps: { value: undefined } });
-        });
-      }
-    } else {
-      this._triggerResize();
     }
 
-    const valueLink = this.props.valueLink;
-    if (typeof valueLink === 'object' && typeof valueLink.requestChange === 'function') {
-      valueLink.requestChange(value);
+    if (has(this.props, 'valueLink')) {
+      this.props.valueLink.requestChange(value);
+    } else if (has(this.props, 'defaultValue')) {
+      this.setState({
+        inputValue: value
+      });
     }
 
     if (this.props.onUpdate) {
       this.props.onUpdate({ value: value });
     }
+    this._triggerResize();
   }
```

Comment:
I think _triggerResize should be called only after all value updates are done. What do you suggest Nik.